sphinx/source/docs/user_guide/annotations.rst

```diff
@@ -74,79 +88,97 @@ keyword argument:
 
     p.circle('x', 'y', legend_label="some label")
 
-If multiple glyphs are given the same label, all the glyphs will be combined into a
-single legend item with that label.
+If you assign the same label name to multiple glyphs, all the glyphs will be
+combined into a single legend item with that label.
 
 .. bokeh-plot:: docs/user_guide/examples/plotting_legend_label.py
     :source-position: above
 
-Automatic Grouping (Python)
+.. _userguide_plotting_legends_legend_group:
+
+Automatic grouping (Python)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-It is often desirable to generate multiple legend items by grouping the values
-in a data source column. It is possible for Bokeh to perform such a grouping by
-passing the ``legend_group`` keyword argument to a glyph method:
+If your data is in the form of a :ref:`ColumnDataSource <userguide_data_cds>`,
+Bokeh can generate legend entries from strings in one of the ColumnDataSource's
+columns. This way, you can create legend entries based on groups of glyphs.
+
+To use data from a column of a ColumnDataSource to generate your plot's legend,
+pass the column name as the ``legend_group`` keyword argument to a glyph method:
 
 .. code-block:: python
 
     p.circle('x', 'y', legend_group="colname", source=source)
 
-When this method is used, the grouping is performed immediately in Python, and
-subsequent Python code will be able to see the individual legend items in
-``Legend.items`` property. If desired, these items can be re-arranged or modified.
+Because ``legend_group`` references a column of a ColumnDataSource, you need to
+always provide a ``source`` argument to the glyph method as well. Additionally,
+the column containing the label names has to be present in the data source at
+that point:
 
 .. bokeh-plot:: docs/user_guide/examples/plotting_legend_group.py
     :source-position: above
 
-.. note::
+Using ``legend_group`` means that Bokeh groups the legend entries immediately.
+Therefore, any subsequent Python code will be able to see the individual legend
+items in the ``Legend.items`` property. This way, you can re-arrange or modify
+the legend at any time.
 
-    To use this feature, a ``source`` argument *must also be provided* to the
-    glyph method. Additionally, the column to be grouped must already be present
-    in the data source at that point.
+.. _userguide_plotting_legends_legend_field:
 
-Automatic Grouping (Browser)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Automatic grouping (browser-side)
```

Comment:
If we say "browser-side" here should we say "Python-side" above to have parallel descriptions? Ive never quite been happy with these parentheticals. The important distinction is 

* once, up-front, before the output is generated or seen in a browser, will not update automatically if data changes

vs

* dynamic, updated in JS in the browser if data changes

But I am not how to succinctly summarize those in teh space of a section heading.

Reply:
I can't think of a good term that would summarize the difference in the headline either. But how about we change the headlines to "Python-side" and "JavaScript-side", for consistency's sake? I think this would make it quite clear that one refers to the Python side and the other refers to the things that happen with JavaScript (i.e. in the browser).